File: app/src/main/java/me/imzack/app/end/injector/module/PlanDetailPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.PlanDetailViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class PlanDetailPresenterModule {

    private final PlanDetailViewContract mPlanDetailViewContract;
    private final int mPlanListPosition;

    public PlanDetailPresenterModule(PlanDetailViewContract planDetailViewContract, int planListPosition) {
        mPlanDetailViewContract = planDetailViewContract;
        mPlanListPosition = planListPosition;
    }

    @Provides
    PlanDetailViewContract providePlanDetailViewContract() {
        return mPlanDetailViewContract;
    }

    @Provides
    int providePlanListPosition() {
        return mPlanListPosition;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/PlanDetailPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.PlanDetailViewContract

@Module
class PlanDetailPresenterModule(private val mPlanDetailViewContract: PlanDetailViewContract, private val mPlanListPosition: Int) {

    @Provides
    fun providePlanDetailViewContract() = mPlanDetailViewContract

    @Provides
    fun providePlanListPosition() = mPlanListPosition
}
